packages/react-components/react-motion/src/factories/createMotionComponent.ts

```diff
@@ -19,21 +19,29 @@ export type MotionComponentProps = {
  *
  * @param value - A motion definition.
  */
-export function createMotionComponent(value: AtomMotion | AtomMotion[] | AtomMotionFn) {
-  const Atom: React.FC<MotionComponentProps> = props => {
-    const { children, imperativeRef } = props;
+export function createMotionComponent<MotionParams extends Record<string, MotionParam> = {}>(
+  value: AtomMotion | AtomMotion[] | AtomMotionFn<MotionParams>,
+) {
+  const Atom: React.FC<MotionComponentProps & MotionParams> = props => {
+    const { children, imperativeRef, ..._rest } = props;
+    const params = _rest as unknown as MotionParams;
     const child = getChildElement(children);
 
     const handleRef = useMotionImperativeRef(imperativeRef);
     const elementRef = React.useRef<HTMLElement>();
+    const paramsRef = React.useRef<MotionParams>(params);
 
     const isReducedMotion = useIsReducedMotion();
 
+    useIsomorphicLayoutEffect(() => {
+      paramsRef.current = params;
```

Comment:
We could have a comment explaining the benefit of putting params in a ref (i.e. changing a param won't interrupt the motion, but will take effect the next time it runs).